runbotics/runbotics-scheduler/src/websocket/bot/bot-lifecycle.service.ts

```diff
@@ -1,62 +1,110 @@
     
     private readonly logger = new Logger(BotLifecycleService.name);
 
-    async handleInterruptedProcessInstanceExecution(bot: IBot): Promise<void> {
+    async handleProcessInstanceInterruption(bot: IBot): Promise<void> {
         const disconnectedBotProcessInstances = await this.processInstanceService.findAllByBotId(bot.id);
         disconnectedBotProcessInstances.forEach(async (processInstance) => {
             if (
                 processInstance.status !== ProcessInstanceStatus.INITIALIZING &&
                 processInstance.status !== ProcessInstanceStatus.IN_PROGRESS
             ) return;
 
-            const newProcessInstance = { 
-                ...processInstance,
-                status: ProcessInstanceStatus.ERRORED,
-                error: 'Bot has been shut down',
+            const newProcessInstance: IProcessInstance = {
+                ...processInstance, 
+                status: ProcessInstanceStatus.ERRORED, 
+                error: 'Bot has been shut down'
             };
 
-            await this.processInstanceService.save(newProcessInstance)
-                .then(() => {
-                    this.logger.log(
-                        `<= Success: process-instance (${
-                            newProcessInstance.id
-                        }) updated after bot disconnection | status: ${
-                            newProcessInstance.status
-                        }, error message: "${
-                            newProcessInstance.error
-                        }"`
-                    );
-                })
-                .catch((err) => {
-                    this.logger.log(`<= Error: process-instance (${newProcessInstance.id}) update after bot disconnection failed | Error: ${err}`);
-                });
-
-            this.handleInterruptedProcessInstanceEvent(bot, processInstance.id);
+            await this.updateInterruptedProcessInstance(newProcessInstance, processInstance.id);            
+
+            await this.handleProcessInstanceEventInterruption(processInstance, bot);
         });
     }
 
-    private async handleInterruptedProcessInstanceEvent (bot: IBot, processInstanceId: string): Promise<void> {
-        const activeProcessInstanceEvents = this.processInstanceEventService.findActiveByProcessInstanceId(processInstanceId);
-        const newProcessInstanceEvent = {
-            ...activeProcessInstanceEvents,
-            status: ProcessInstanceEventStatus.ERRORED,
-            error: 'Bot has been shut down',
-        };
+    private async handleProcessInstanceEventInterruption (processInstance: IProcessInstance, bot: IBot): Promise<void> {
+        const activeEvents: IProcessInstanceEvent[] = await this.processInstanceEventService.findActiveByProcessInstanceId(processInstance.id);
+        const processId: number = await this.processInstanceService.findById(processInstance.id).then(processInstance => processInstance.process.id);
+        
+        if(!activeEvents) return;
```

Comment:
this guard should be higher, redundant DB call for processId

Reply:
done